tests/pos/i12715/full.scala

```diff
@@ -0,0 +1,41 @@
+package repro
+
+import compiletime.{constValue, erasedValue}
+
+sealed trait ValidateExprInt
+
+class And[A <: ValidateExprInt, B <: ValidateExprInt] extends ValidateExprInt
+class GreaterThan[T <: Int]                           extends ValidateExprInt
+
+object Repro:
+  inline def validate[E <: ValidateExprInt](v: Int): String =
+    inline val failMsg = validateV[E](v)
+    inline if failMsg == "neverPass" then "neverPass"
```

Comment:
This seems to now have a different error:
```
-- Error: tests/pos/i12715/full.scala:41:59 ------------------------------------
41 |    println(validate[And[GreaterThan[10], GreaterThan[12]]](5))
   |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |Cannot reduce `inline if` because its condition is not a constant value: failMsg.==("neverPass")
   | This location contains code that was inlined from full.scala:13
1 error found
```
It looks like a related issue :/

Reply:
Or is it an issue @nicolasstucki  ?

Reply:
After checking the example, it does seem to fail correctly. I will fix it or remove it.

Reply:
We should minimize this example and open another issue.

Reply:
This was failing correctly, the example wasn't fully working even after the fix here.